main/profile.py

```diff
@@ -112,6 +112,7 @@ def profile_password():
         errors = True
 
       if not (form.errors or errors):
+        user_db.password_salt = util.password_salt()
```

Comment:
you could simply use `util.uuid()` whenever you need a salt or anything random... :)